Tests/Pcap++Test/main.cpp

```diff
@@ -5259,37 +5293,97 @@ PTF_TEST_CASE(TestTcpReassemblyIPv6_OOO)
 
+	RawPacket lastPacket = packetStream.back();
+
+	packetStream.pop_back();
+
+	for(std::vector<RawPacket>::iterator iter = packetStream.begin(); iter != packetStream.end(); iter++)
+	{
+		Packet packet(&(*iter));
+		tcpReassembly.reassemblePacket(packet);
+	}
+
+	TcpReassembly::ConnectionInfoList managedConnections = tcpReassembly.getConnectionInformation(); // make a copy of list
+	PTF_ASSERT(managedConnections.size() == 3, "Size of managed connection list isn't 3");
+	PTF_ASSERT(results.flowKeysList.size() == 3, "Num of flow keys isn't 3");
+
+	TcpReassembly::ConnectionInfoList::const_iterator iterConn1 = managedConnections.find(results.flowKeysList[0]);
+	TcpReassembly::ConnectionInfoList::const_iterator iterConn2 = managedConnections.find(results.flowKeysList[1]);
+	TcpReassembly::ConnectionInfoList::const_iterator iterConn3 = managedConnections.find(results.flowKeysList[2]);
+	PTF_ASSERT(iterConn1 != managedConnections.end(), "Connection #1 not found");
+	PTF_ASSERT(iterConn2 != managedConnections.end(), "Connection #2 not found");
+	PTF_ASSERT(iterConn3 != managedConnections.end(), "Connection #3 not found");
+	PTF_ASSERT(tcpReassembly.isConnectionOpen(iterConn1->second) == 0, "Connection #1 is still open");
+	PTF_ASSERT(tcpReassembly.isConnectionOpen(iterConn2->second) == 0, "Connection #2 is still open");
+	PTF_ASSERT(tcpReassembly.isConnectionOpen(iterConn3->second) == 0, "Connection #3 is still open");
+
+	PCAP_SLEEP(2);
+
+	tcpReassembly.reassemblePacket(&lastPacket); // automatic cleanup of 1 item
+	PTF_ASSERT(tcpReassembly.getConnectionInformation().size() == 2, "Size of managed connection list isn't 2");
+
+	tcpReassembly.purgeClosedConnections(); // manually initiated cleanup of 1 item
+	PTF_ASSERT(tcpReassembly.getConnectionInformation().size() == 1, "Size of managed connection list isn't 1");
+
+	tcpReassembly.purgeClosedConnections(time(NULL), 0xFFFFFFFF); // manually initiated cleanup of all items
+	PTF_ASSERT(tcpReassembly.getConnectionInformation().size() == 0, "Size of managed connection list isn't zero");
+
+	const TcpReassemblyMultipleConnStats::FlowKeysList &flowKeys = results.flowKeysList;
+	iterConn1 = managedConnections.find(flowKeys[0]);
+	iterConn2 = managedConnections.find(flowKeys[1]);
+	iterConn3 = managedConnections.find(flowKeys[2]);
+	PTF_ASSERT(iterConn1 != managedConnections.end(), "Connection #1 not found in flow keys list");
+	PTF_ASSERT(iterConn2 != managedConnections.end(), "Connection #2 not found in flow keys list");
+	PTF_ASSERT(iterConn3 != managedConnections.end(), "Connection #3 not found in flow keys list");
+	PTF_ASSERT(tcpReassembly.isConnectionOpen(iterConn1->second) == -1, "Connection #1 still exists");
+	PTF_ASSERT(tcpReassembly.isConnectionOpen(iterConn2->second) == -1, "Connection #2 still exists");
+	PTF_ASSERT(tcpReassembly.isConnectionOpen(iterConn3->second) == -1, "Connection #3 still exists");
 }

```

Comment:
Since this is a new test case, you can use the new assertion macros I recently added: `PTF_ASSERT_EQUAL`, `PTF_ASSERT_TRUE`, `PTF_ASSERT_FALSE`, `PTF_ASSERT_NULL`, `PTF_ASSERT_NOT_NULL` and `PTF_ASSERT_BUF_COMPARE`. 

Please take a look at the new test cases that are already using them: https://github.com/seladb/PcapPlusPlus/blob/220434db398e68b5fc5ae473bbb439a61f485dda/Tests/Packet%2B%2BTest/main.cpp#L7093

Eventually I'd like to migrate all test cases to use them, but it will probably be a long process. In the meantime, let's try to use them in the new test cases we're writing

Reply:
Unfortunately not yet. You can leave this one as it is